master/internal/api_tasks.go

```diff
@@ -373,6 +375,66 @@ func (a *apiServer) TaskLogs(
 	})
 }
 
+func (a *apiServer) Monitor(ctx context.Context, taskID string, logs []*model.TaskLog) error {
+	// TODO(ft) do all logs have same taskID? Or should we just assume that they don't need to.
+	// TODO(ft) do all logs stream through here? K8S I think we are planning to add log through
+	// k8s api too.
+
+	isExp, exp, err := expFromTaskID(ctx, model.TaskID(taskID))
+	if err != nil {
+		return err
+	}
+	if isExp == false {
+		return nil
+	}
+
+	activeConfig, err := a.m.db.ActiveExperimentConfig(exp.ID)
+	if err != nil {
+		return err
+	}
+
+	for _, l := range logs {
+		if l.AgentID == nil {
+			return fmt.Errorf("agentID must be non nil") // TODO can we get away with this?
+			// It feels kinda annoying to let our database have a possible null that theoretically
+			// shouldn't happen.
+		}
+
+		for _, lpp := range activeConfig.LogPatternPolicies() {
+			regex := fmt.Sprintf("(.*)%s(.*)", lpp.Pattern())
```

Comment:
question: do you mean to use `strings.Contains(l.Log, regex)` to match pattern? If this is the right [doc](https://pkg.go.dev/strings#Contains), it only supports substr lookup?

Reply:
Hmm, I guess what I'm doing is the same as substr lookup.